jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -259,6 +264,16 @@ private Span setTagAsObject(String key, Object value) {
     return this;
   }
 
+  /**
+   * uses {@link #setTagAsObject} to only insert tags that don't have special significance
+   */
+  private void sanitizeAndSetTags(Map<String, Object> tags) {
```

Comment:
That incorrect, we check and discard tags. 
For e.g., one can never set a `span.kind:server` on the `Span` object. It is always discarded after setting the `isRPC` field to true. Other tags that are discarded are PEER_PORT, PEER_HOST, PEER_SERVICE, etc.